rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,143 @@
+
+An issue has been raised that the logic for monitoring/validating the health of a network is orthogonal to deployment, which is understandable. The longer term plan of eventually transitioning to go makes the possible value of this sort of command much higher, and as such a bash implementation is less worthwhile. The grafana dashboard + gcp UI url for a given testnet can just as easily be returned by the start/restart/upgrade commands.
+
+## Beyond the minaform tool
+
+### Minaform Docker image
+
+To address pain points #7 and #8, and to some extent the larger problem of how to make it easy for anyone, from anywhere, to deploy networks, we need a minaform docker image that can be bundled alongside all of its dependencies (nodejs for coda-network, some of our deb packages like generate-keypair, terraform, helm, most of mina-automation repo, and the helm charts themselves). These dependencies can then be handled by updates to the docker image, and anyone having trouble with version mismatches can mount their credentials into a docker container and get moving quickly.
+
+In addition to use on any developer laptop, the docker image could be deployed directly into the cluster, google cloud shell, or a google cloud VM to allow for low-latency deployment to gcloud even from a high-latency network (hope that helps @bkase @mrmr1993 !).
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+
+
+## Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Why Bash?
+
+* Key generation and upload are handled by bash scripts as of now so this will allow for quick iteration as hacky internals are removed, without changing the command line interface or the user experience. This design is intended to not introduce any additional tools to the process that we were not already committed to, so that after the minimum viable verison is in place we can re-evaluate if the system needs new tools.
+
+Why is this design the best in the space of possible designs?
+
+* To a point mentioned in the comments, bash has plenty of built-in support for properly parsing command-line arguments (see scripts/release-docker.sh in the mina repo for an example of handling that more properly), and very little needs to be done to glue terraform + coda-network together besides standardizing file paths and manipulating strings so the outputs of one tool properly match the inputs of another. A carefully written and commented bash script is the shortest path to a usable solution, and when the requirements for a solution exceed the plan outlined in this RFC we can just as easily continue from this point while rewriting in a compiled and typechecked language.
```

Comment:
Plenty of support is a big statement for `bash`. Maybe a little too big. See previous comments but thinking this is where the readability/writeability woes don't help its case. 

If we had more experience/examples throughout src, my vote would be for something like `javascript` which could help us get away from CLIs all together and into the modern, less noisy, more usable/interactive world of web interfaces using Javascript.